File: coding_interviews/leetcode/easy/single_number/single_number.py
# https://leetcode.com/problems/single-number

def single_number(nums):
    counter = {}
    
    for num in nums:
        if num in counter:
            counter[num] += 1
        else:
            counter[num] = 1
            
    for num in nums:
        if counter[num] == 1: return num

File: coding_interviews/leetcode/easy/single-number/single-number.js
function singleNumber(nums) {
  let hashmap = {};

  for (let num of nums) {
    if (hashmap[num]) {
      hashmap[num]++;
    } else {
      hashmap[num] = 1;
    }
  }

  for (let num of nums) {
    if (hashmap[num] === 1) {
      return num;
    }
  }
}
